### ISSUE TITLE
Break apart monolithic TextStyle component

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#5519]
This should probably be done in a future PR just to not let this one grow too much, but doing this for TextStyle should probably be done too.

_Originally posted by @IceSentry in https://github.com/bevyengine/bevy/issues/5513#issuecomment-1200472999_

Sibling to #5511, and should be done for the same reasons.

**Comment by Nilirad** [comment]
Pasting what I said in #5511:

> Not sure if in-topic, but I think it would be good UX to have font weight/style and text decoration as their own components:

> - `Underline` (as or with an enum with `Normal`, `Dotted`, `Wavy`, etc variants)
> - `Overline`
> - `LineThrough`
> - `Bold`
> - `Italic`

**Comment by rparrett** [comment]
I'm slightly confused by this issue -- I don't think `TextStyle` has been used as a Component since #1245 (before this issue was created?) so I'm not sure how breaking it up would make things more ergonomic.

This could potentially become relevant again if "text sections as entities" gains traction in the future.

**Comment by alice-i-cecile** [comment]
Yeah, sorry about that, the wording was definitely misleading. I think that "text styles as entities" is still an interesting direction, but this isn't super actionable as is.
